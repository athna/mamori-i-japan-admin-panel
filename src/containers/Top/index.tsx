import React, { useCallback, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import ChartLine from '../../components/ChartLine';

import { getAnalytics } from '../../redux/Analytics/actions';

export default () => {
  const dispatch = useDispatch();
  const { lineChartData } = useSelector((store: any) => store.analytics);

  const fetchData = useCallback(() => dispatch(getAnalytics()), [dispatch]);

  useEffect(() => {
    fetchData();
  }, [fetchData]);

  return (
    <div>
      {lineChartData.length && (
        <ChartLine
          dataKey={'month'}
          fields={['Tokyo', 'London']}
          sourceData={lineChartData}
        />
      )}
    </div>
  );
};
