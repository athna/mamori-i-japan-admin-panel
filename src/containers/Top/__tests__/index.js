import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { configure, shallow } from 'enzyme';
import { render } from '@testing-library/react';
import Adapter from 'enzyme-adapter-react-16';

import Top from '../';

configure({ adapter: new Adapter() });

describe('Top', () => {
  const initialState = { lineChartData: [] };
  const mockStore = configureStore();
  let store;
  it('should render correctly with no props', () => {
    store = mockStore(initialState);

    const component = shallow(
      <Provider store={store}>
        <Top />
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });

  // it('should render correctly content: "top"', () => {
  //   const { getByText } = render(<Top />);
  //   const linkElement = getByText(/top/i);

  //   expect(linkElement).toBeInTheDocument();
  // });
});
