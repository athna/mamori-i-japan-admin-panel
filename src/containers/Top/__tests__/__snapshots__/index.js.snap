// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top should render correctly with no props 1`] = `
<div>
  <_default
    dataKey="month"
    fields={
      Array [
        "Tokyo",
        "London",
      ]
    }
    sourceData={
      Array [
        Object {
          "London": 3.9,
          "Tokyo": 7,
          "month": "Jan",
        },
        Object {
          "London": 4.2,
          "Tokyo": 6.9,
          "month": "Feb",
        },
        Object {
          "London": 5.7,
          "Tokyo": 9.5,
          "month": "Mar",
        },
        Object {
          "London": 8.5,
          "Tokyo": 14.5,
          "month": "Apr",
        },
        Object {
          "London": 11.9,
          "Tokyo": 18.4,
          "month": "May",
        },
        Object {
          "London": 15.2,
          "Tokyo": 21.5,
          "month": "Jun",
        },
        Object {
          "London": 17,
          "Tokyo": 25.2,
          "month": "Jul",
        },
        Object {
          "London": 16.6,
          "Tokyo": 26.5,
          "month": "Aug",
        },
        Object {
          "London": 14.2,
          "Tokyo": 23.3,
          "month": "Sep",
        },
        Object {
          "London": 10.3,
          "Tokyo": 18.3,
          "month": "Oct",
        },
        Object {
          "London": 6.6,
          "Tokyo": 13.9,
          "month": "Nov",
        },
        Object {
          "London": 4.8,
          "Tokyo": 9.6,
          "month": "Dec",
        },
      ]
    }
  />
</div>
`;
